tests/stress/deployments.py

```diff
@@ -0,0 +1,761 @@
+import os
+import shutil
+import subprocess
+import time
+from abc import ABC
+from typing import List
+
+import yaml
+from gqlalchemy import Memgraph
+from neo4j import GraphDatabase
+
+# paths
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+BASE_DIR = os.path.normpath(os.path.join(SCRIPT_DIR, "..", ".."))
+BUILD_DIR = os.path.join(BASE_DIR, "build")
+
+
+class Deployment(ABC):
+    def __init__(self):
+        super().__init__()
+
+    def start_memgraph(self, additional_flags: List[str] = None):
+        pass
+
+    def stop_memgraph(self) -> None:
+        pass
+
+    def cleanup(self) -> None:
+        pass
+
+    def wait_for_server(self, port=None) -> None:
+        pass
+
+    def execute(self, query: str) -> None:
+        memgraph = Memgraph()
+        memgraph.execute(query)
+
+    def execute_and_fetch(self, query: str) -> None:
+        memgraph = Memgraph()
+        return memgraph.execute_and_fetch(query)
+
+    def execute_for_tenant(self, tenant: str, query: str) -> None:
+        if self._driver is None:
+            self._driver = GraphDatabase.driver("bolt://127.0.0.1:7687", auth=("", ""))
+        with self._driver.session(database=tenant) as session:
+            session.run(query)
+
+
+class DefaultStandaloneDeployment(Deployment):
+    def __init__(self, memgraph: str, flags: List[str]):
+        super().__init__()
+        self._memgraph = memgraph
+        self._flags = flags
+
+    def start_memgraph(self, additional_flags: List[str] = None):
+        """Starts Memgraph and return the process"""
+        cwd = os.path.dirname(self._memgraph)
+
+        flags = [] if additional_flags is None else additional_flags
+
+        cmd = [self._memgraph] + self._flags + flags
+        proc = subprocess.Popen(cmd, cwd=cwd)
+        self.wait_for_server(7687)
+
+        assert proc.poll() is None, "The database binary died prematurely!"
+
+        self._memgraph_proc = proc
+
+    def stop_memgraph(self) -> None:
+        self._memgraph_proc.terminate()
+        ret_mg = self._memgraph_proc.wait()
+        if ret_mg != 0:
+            raise Exception("Memgraph binary returned non-zero ({})!".format(ret_mg))
+
+    def cleanup(self):
+        if self._memgraph_proc.poll() != None:
+            return
+        self._memgraph_proc.kill()
+        self._memgraph_proc.wait()
+
+    def wait_for_server(self, port) -> None:
+        cmd = ["nc", "-z", "-w", "1", "127.0.0.1", str(port)]
+        while subprocess.call(cmd) != 0:
+            time.sleep(0.5)
+        time.sleep(2)
+
+
+class DefaultHADeployment(Deployment):
+    def __init__(self, memgraph: str, flags: List[str]):
```

Comment:
maybe a good configuration point would be to have a number of replicas and a number of coordinators you want to use. Not necessary to add immediately but different clients will want to have different number of instances